shared/constants/transaction.js

```diff
@@ -0,0 +1,95 @@
+// Transaction Type is a MetaMask construct used internally
+// There are three types at present
+// 1. `'standard'` - A standard transaction, usually the first with a given nonce
+// 2. `'cancel'`   - A transaction submitted with the same nonce as a previous
+//                   transaction, a higher gas price and a zeroed out send amount.
+//                   Useful for users who accidentally send to erroneous addresses
+//                   or if they send too much.
+// 3. `'retry'`    - When a transaction is failed it can be retried by resubmitting
+//                   the same transaction with a higher gas fee. This type is also
+//                   used to speed up pending transactions. This is accomplished by
+//                   creating a new tx with the same nonce and higher gas fees.
+export const TRANSACTION_TYPE_STANDARD = 'standard'
+export const TRANSACTION_TYPE_CANCEL = 'cancel'
+export const TRANSACTION_TYPE_RETRY = 'retry'
+
+// Transaction Status is a mix of Ethereum terminology and MetaMask construct
+// 1. `'unapproved'` - A new transaction that the user has not approved or rejected
+// 2. `'approved'`   - The user has approved the transaction in the MetaMask UI
+// 3. `'rejected'`   - The user has rejected the transaction in the MetaMask UI
+// 4. `'signed'`     - The transaction has been signed
+// 5. `'submitted'`  - The transaction has been submitted
+// 6. `'failed'`     - The transaction has failed for some reason
+// 7. `'dropped'`    - The transaction was dropped due to a tx with same nonce being accepted
```

Comment:
Is that the only reason a transaction would be dropped? :thinking: I thought the network could drop it for other reasons as well.

Reply:
Yeah lots of other reasons. I'm going to update this (and updating my own knowledge)